client/state/push-notifications/actions.js

```diff
@@ -0,0 +1,324 @@
+	return ( dispatch, getState ) => {
+		const state = getState();
+		const lastUpdated = getLastUpdated( state );
+		debug( 'Subscription last updated: ' + lastUpdated );
+
+		let age;
+
+		if ( lastUpdated ) {
+			age = moment().diff( moment( lastUpdated ), 'days' );
+			if ( age < DAYS_BEFORE_FORCING_REGISTRATION_REFRESH ) {
+				debug( 'Subscription did not need updating.', age );
+				return;
+			}
+		}
+
+		debug( 'Subscription needed updating.', age );
+
+		const sub = getSavedSubscription( state );
+		if ( ! sub ) {
+			debug( 'No subscription to send to WPCOM' );
+			// @TODO dispatch something :)
+			return;
+		}
+		debug( 'Sending subscription to WPCOM', sub );
+
+		wpcom.undocumented().registerDevice( JSON.stringify( sub ), 'browser', 'Browser' )
+			.then( data => dispatch( {
+				type: PUSH_NOTIFICATIONS_RECEIVE_REGISTER_DEVICE,
+				data
+			} ) )
+			.catch( err => debug( 'Couldn\'t register device', err ) )
+		;
+	};
+}
+
+export function activateSubscription() {
+	return dispatch => {
+		window.navigator.serviceWorker.ready
+			.then( serviceWorkerRegistration => {
+				serviceWorkerRegistration.pushManager.subscribe( { userVisibleOnly: true } )
+					.then( subscription => {
+						dispatch( receiveSubscription( subscription ) );
+						dispatch( checkPermissionsState() );
+					} )
+					.catch( err => {
+						debug( 'Error receiving subscription', err );
+						dispatch( block() );
+					} )
+				;
+			} )
+			.catch( err => debug( 'Error activating subscription', err )	)
+		;
+	};
+}
+
+export function unregisterDevice() {
+	return ( dispatch, getState ) => {
+		const deviceId = getDeviceId( getState() );
+		if ( ! deviceId ) {
+			debug( 'Couldn\'t unregister device. Unknown device ID' );
+			dispatch( receiveUnregisterDevice( {} ) );
+			return;
+		}
+		wpcom.undocumented().unregisterDevice( deviceId )
+			.then( ( data ) => {
+				debug( 'Successfully unregistered device', data );
+				dispatch( receiveUnregisterDevice( data ) );
+			} )
+			.catch( err => {
+				debug( 'Couldn\'t unregister device', err );
+				dispatch( receiveUnregisterDevice( {} ) );
+			} );
+	};
+}
+
+export function receiveUnregisterDevice( data ) {
+	return {
+		type: PUSH_NOTIFICATIONS_RECEIVE_UNREGISTER_DEVICE,
+		data
+	};
+}
+
+export function checkPermissionsState() {
+	return dispatch => {
+		window.navigator.serviceWorker.ready
+			.then( ( serviceWorkerRegistration ) => {
+				serviceWorkerRegistration.pushManager.permissionState( { userVisibleOnly: true } )
+					.then( permissionState => {
+						debug( 'Received push messaging state', permissionState );
+						dispatch( receivePermissionState( permissionState ) );
+					} )
+					.catch( err => {
+						debug( 'Error checking permission state', err );
+						dispatch( receivePermissionState( 'denied', err ) );
+					} )
+				;
+			} )
+			.catch( err => debug( 'Error checking permission state -- not ready', err )	)
+		;
+	};
+}
+
+export function serviceWorkerLoadingError( err ) {
```

Comment:
This is only called one place, just move the 2 relevant lines in the place of the caller.